test/cpp_extensions/open_registration_extension/torch_openreg/third_party/openreg/csrc/stream.cpp

```diff
@@ -267,6 +270,9 @@ orError_t orStreamWaitEvent(orStream_t stream, orEvent_t event, unsigned int) {
   if (!stream || !event)
     return orErrorUnknown;
 
+  if (event->impl->device_index != stream->device_index)
```

Comment:
`OpenReg` is a CUDA-like accelerator, so I suppose it's time to refactor the error types and error messages to align with CUDA, making it easier for developers to find the root cause of failures.

We can do this in follow-up PR.